Task5

lemonclouuds opened the pull request.

lemonclouuds committed:
Implemented checkout page & OrderDao

lemonclouuds committed:
Implemented order overview

lemonclouuds committed:
Implemented Dos Filter

lemonclouuds committed:
Modified filter to check for 20 requests/minute

lemonclouuds committed:
Added tests

lemonclouuds committed:
Fixed tags for checkout & order overview

Alexandra-ES commented on src/main/java/com/es/phoneshop/model/product/order/OrderNotFoundException.java:
```diff
@@ -0,0 +1,16 @@
+package com.es.phoneshop.model.product.order;
+
+public class OrderNotFoundException extends RuntimeException {
+    Long id;
```
private

Alexandra-ES commented on src/main/java/com/es/phoneshop/model/product/order/OrderNotFoundException.java:
```diff
@@ -0,0 +1,16 @@
+    }
+
+    public OrderNotFoundException(Long id) {
+        this.id = id;
```
add super with some message, for example super("Order not found " + id);

Alexandra-ES commented on src/main/java/com/es/phoneshop/web/CheckoutPageServlet.java:
```diff
@@ -0,0 +1,106 @@
+        if (value == null || value.isEmpty()) {
+            errors.put("deliveryDate", "Value is required");
+        } else {
+            order.setDeliveryDate(LocalDate.parse(value, DateTimeFormatter.ofPattern(pattern)));
```
add validation for past date

Alexandra-ES commented on src/main/java/com/es/phoneshop/web/CheckoutPageServlet.java:
```diff
@@ -0,0 +1,106 @@
+        if (value == null || value.isEmpty()) {
+            errors.put("paymentMethod", "Value is required");
+        } else {
+            order.setPaymentMethod(PaymentMethod.valueOf(value));
```
[opt] return nice formatted values

Alexandra-ES commented on src/main/webapp/WEB-INF/pages/orderOverview.jsp:
```diff
@@ -0,0 +1,89 @@
+  </table>
+  <h2>Your details</h2>
+  <table class="customerDetails">
+    <tags:orderOverview name="firstName" label="Firs tname" order="${order}"/>
```
[opt] change the name of tag to a more descriptive one, for example orderOverviewRow

Alexandra-ES commented on src/main/java/com/es/phoneshop/web/OrderOverviewPageServlet.java:
```diff
@@ -0,0 +1,30 @@
+import java.io.IOException;
+
+public class OrderOverviewPageServlet extends HttpServlet {
+    private OrderDao orderDao;
```
use OrderService here

lemonclouuds committed:
Fix

lemonclouuds committed:
Fix
-added phone number validation
-fixed 500 error page
-using OrderService instead of OrderDao

Alexandra-ES commented on src/main/java/com/es/phoneshop/web/CheckoutPageServlet.java:
```diff
@@ -51,6 +51,7 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         setRequiredParameter(request, "firstName", errors, order::setFirstName);
         setRequiredParameter(request, "lastName", errors, order::setLastName);
         setRequiredParameter(request, "phone", errors, order::setPhone);
```
remove this line

lemonclouuds committed:
Removed redundant line
